### PR TITLE
Cleanup disks service

### LINKED ISSUE #1001: Clean up ./cloud/services
/kind cleanup

**Describe the solution you'd like**
A while back, the code used to interact with Azure under `./cloud/services` was interfaced to allow some separation of concern between the controller / scopes and the types from the Azure SDK. With the advent of the `azure.Service` interface, Azure SDK / client code doesn't leak into the controllers. In fact, there should not longer be dependencies on the code that directly interacts with Azure. We should use this as an opportunity to crystallize this design goal of refactoring to `azure.Service` by cleaning up the service internals.

Current structure:
```
cloud/services/scalesets/
├── client.go
├── mock_scalesets
│   ├── client_mock.go
│   ├── doc.go
│   └── scalesets_mock.go
├── service.go
├── vmss.go
└── vmss_test.go
```

Proposed structure:
```
cloud/services/scalesets/
├── client.go  <-- don't export this; make it private
├── mock_scalesets
│   ├── client_mock.go
│   ├── doc.go
│   └── scaleset_mock.go
├── scaleset.go <-- includes service and scope
└── scaleset_test.go
```
The only thing exported from the new structure is the `Service` struct and the `Scope` interface for each service. This will ensure that the encapsulation will not be violated and guide future contributors.

### PR DESCRIPTION
**Pull request by shysank**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Cleanup disks service
- Make `Client` interface private ( the methods are still exported to generate mocks)
- Move `DiskScope` and `Service` struct to `disks.go`
- `NewService()` -> `New()` 
- Remove redundant initialization code in `disks_test.go`

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1001 (Partially)

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Cleanup disks service
```

### CODE REVIEW

**Comment by nader-ziada** [comment]
I like the approach 👍

**Comment by shysank** [comment]
API Diff
```
sigs.k8s.io/cluster-api-provider-azure/cloud/services/disks/mock_disks
  Incompatible changes:
  - MockClient: removed
  - MockClientMockRecorder: removed
  - NewMockClient: removed
  Compatible changes:
  - Mockclient: added
  - MockclientMockRecorder: added
  - NewMockclient: added
sigs.k8s.io/cluster-api-provider-azure/cloud/services/disks
  Incompatible changes:
  - AzureClient: removed
  - Client: removed
  - NewClient: removed
  - NewService: removed
  - Service.Client: removed
  Compatible changes:
  - New: added
```

**Comment by nader-ziada** [comment]
@devigned what do you think?

File: cloud/services/disks/client.go [hunk 1]
```diff
@@ -25,21 +25,21 @@ import (
...
 }
 
 // AzureClient contains the Azure go-sdk Client
-type AzureClient struct {
+type azureClient struct {
 	disks compute.DisksClient
 }
 
-var _ Client = &AzureClient{}
+var _ client = &azureClient{}
```
Comment (devigned) [inline cloud/services/disks/client.go:37]:
```suggestion
var _ client = (*azureClient)(nil)
```

^ no allocation needed

**Review by devigned — CHANGES_REQUESTED** [review]
It's beautiful!

I left one super small nit. Thank you and great work!

File: cloud/services/disks/client.go (hunk 1 above)
Comment (shysank) [reply cloud/services/disks/client.go]: Fixed it, Thanks!

**Review by CecileRobertMichon — APPROVED** [review]
lgtm

**Review by devigned — APPROVED** [review]
/lgtm
/approve

**Pull request merged**